### DISCUSSION TITLE
Hierarchical access? Limiting information at different levels?

### DISCUSSION THREAD

**Discussion by code-reflex** [getredash/redash#6010] [Q&A]
Redash noob here. I could not easily find the answer to what I am looking for & hence this question.

Requesting your inputs on how this below described scenario can be built in Redash.

Assume that we have data (for example: sales figures, salesman wise) at city office level, rolled up to state level access at state office & similarly available for access at country office level.  I want to allow users to view data  / dashboard / reports only at their permissible level.  E.g.  The Austin office should be able to view data / metrics only pertaining to their office and the Houston office can only see the numbers of their city. Now, at the state level office, they can see the data of Texas state (i.e., includes data of both Houston & Austin). The same way, people at Country level office can access data from across all the states consolidated / with drill down across all the states.

Basic assumption: Data will have relevant city & state / office-code identifiers to segregate / group / join the data. We can also add additonal tables as necessary to build needed mappings between people, customer, city, state & country etc.

How can I restrict data access to users at each level in the above example?   Can you please guide me and / or point me to relevant resources to achieve this please?

Thank you.

**Comment by RageOsja** [comment]
Did you find the answer? because i have the same question

**Reply by code-reflex** [reply]
nope :(
